Add a guide about ZNC (#206)
thelounge/thelounge.github.io: _guides/znc.md

---
layout: documentation
title: Set up ZNC with The Lounge
---

> ## Why you may not need znc

Reviewer: ```suggestion
## Why you may not need ZNC
```

The Lounge already gives you most of the features you expect from a bouncer:

> - Scroll back is already available from within The Lounge, even across restarts, if logging to sqlite is enabled

Reviewer: ```suggestion
- Scrollback is already available from within The Lounge, even across restarts, [if message storage is configured](/docs/configuration#messagestorage)
```

- Multiple devices can connect to The Lounge at the same time, using The Lounge as a client

- Nick, mode changes etc. are kept, as The Lounge never disconnects from the IRC server

- Push notifications are available on Android and on modern browsers

## Setting up ZNC
If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.

In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:

 - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. Works with any client even when they are not znc aware

 - [Playback](https://wiki.znc.in/Playback): IRC clients may request the module to send a partial buffer playback starting from and ending to a certain point of time. Only works with clients that respect the znc specific capability. The Lounge supports this functionality out of the box

> ## Connect to multiple Networks from The Lounge

Reviewer: ```suggestion
## Connect to multiple networks
```

---
layout: documentation
title: Set up ZNC with The Lounge
description: How to configure ZNC for use with The Lounge
---

## Why you may not need ZNC
The Lounge already gives you most of the features you expect from a bouncer:

- Scrollback is already available from within The Lounge, even across restarts, [if message storage is configured](/docs/configuration#messagestorage).

- Multiple devices can connect to The Lounge at the same time, using The Lounge as a client.

- The Lounge keeps a persistent connection to the IRC server even if you don't have the client open anywhere.

- Push notifications are available if they are supported by your platform and browser.

> ## Setting up ZNC

Reviewer 2: ```suggestion
## Set up ZNC
```

To be consistent with the next bullet (and most pages of the website, if I'm correct).

If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.

> In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:

Reviewer 2: ```suggestion
In order for The Lounge to properly integrate with ZNC (and other clients), you may want to use either one of the following modules:
```

Author: I'm going to replace the first 'The Lounge' instance, else the "(and other clients)" doesn't make any sense. Is that ok?

 - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. This works with any client even when they are not ZNC aware.

 - [Playback](https://wiki.znc.in/Playback): IRC clients may request the module to send a partial buffer playback starting from and ending to a certain point of time. Only works with clients that respect the ZNC specific capability. The Lounge supports this functionality out of the box.

## Connect to multiple networks

> For each IRC server connected in ZNC, you will need a separate IRC server connection in The Lounge.

Reviewer 2: > connected in ZNC

Is it just me or "in" sounds off?
Would something else seem better? "through", "via", "over", etc.

Author: That was a change request from @xPaw. The original was "from The Lounge" I believe.
What do you prefer?

Reviewer 3: connected through ZNC

---
layout: documentation
title: Set up ZNC with The Lounge

> description: How to configure ZNC for use with The Lounge

Reviewer 2: ```suggestion
description: Extend the server capabilities of The Lounge using ZNC and its modules
```

---
layout: documentation
title: Set up ZNC with The Lounge
---

## What The Lounge gives you out of the box
The Lounge already gives you most of the features you expect from a bouncer:

- Scroll back is already available from within The Lounge, even across restarts, if logging to sqlite is enabled

- Multiple devices can connect to The Lounge at the same time, using The Lounge as a client

> - Nick, mode changes etc. are kept, as The Lounge never disconnects from the IRC server

Reviewer: ```suggestion
- The Lounge keeps a persistent connection to the IRC server even if you don't have the client open anywhere.
```

> - Push notifications are available on Android and on modern browsers

Reviewer: ```suggestion
- Push notifications are available if they are supported by your platform and browser
```

## Benefits of using ZNC
There are some benefits of using ZNC in addition to The Lounge.

- Use a terminal based IRC client like weechat or irssi on the desktop, but The Lounge on mobile and still share the same nickname on both

- Add push notifications if they aren't supported directly via The Lounge (for example on iOS via the ZNC-push module)

- Have a centralized block list across different clients

- Detach from channels and auto-attach on activity

- Run multiple The Lounge versions, sharing the data (for example stable and dev in order to test the next The Lounge PR)

## Setting up ZNC
If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.

In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:

>  - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. Works with any client even when they are not znc aware

Reviewer: ```suggestion
 - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. This works with any client even when they are not ZNC aware
```

 - [Playback](https://wiki.znc.in/Playback): IRC clients may request the module to send a partial buffer playback starting from and ending to a certain point of time. Only works with clients that respect the znc specific capability. The Lounge supports this functionality out of the box

## Connect to multiple Networks from The Lounge

> For each IRC server connected from ZNC, you will need a separate irc server connection from within The Lounge.

Reviewer: ```suggestion
For each IRC server connected in ZNC, you will need a separate IRC server connection in The Lounge.
```

---
layout: documentation
title: Set up ZNC with The Lounge
---

## What is ZNC?
ZNC is an advanced IRC bouncer, which inevitably begs the question, what's a bouncer?

Usually an IRC client connects directly to an IRC server. An IRC bouncer sits between the client (for example thelounge) and the irc server (for example Freenode).
That allows it to add some additional functionality over a direct connection, some examples listed below

- Attach multiple clients (say a computer, phone, tablet etc.) to the bouncer, all sharing the same nick.

- Messages are saved even when no client is connected, playing back what was said when a client comes back

- As the bouncer (preferably) never disconnects, mode changes, nick names and channels are kept at all times

- Sharing block lists / using additional functionality across different clients, without having to configure each one of them

## What thelounge gives you out of the box

> Thelounge already gives you most of the features you expect from a bouncer:

Reviewer 3: The Lounge (brand name)

Reviewer 3: also everywhere else it's used

Author: yeah, just wanted to mention that I'll fix it everywhere
